File: go/one_away.go
package main

import (
  "fmt"
  "math"
)


func removeSymbol(s string, index int) string {
  return s[:index] + s[index + 1:]
}


func checkEdit(s1 string, s2 string) bool {
  length1 := float64(len(s1))
  length2 := float64(len(s2))

  if math.Abs(length1 - length2) >= 2 {
    return false
  }

  length := math.Min(length1, length2)

  i := 0
  for i < int(length) {
    if s1[i] == s2[i] {
      i++
    } else {
      if length1 == length2 {
        s1, s2 = removeSymbol(s1, i), removeSymbol(s2, i)
        length--
      } else if length1 > length2{
        s1 = removeSymbol(s1, i)
      } else if length1 < length2{
        s2 = removeSymbol(s2, i)
      }

      if s1 != s2{
        return false
      }
    }
  }

  return true
}


func main() {
  str1 := "palinr"
  str2 := "palidr"

  fmt.Println(checkEdit(str1, str2))
}

File: python/one_away.py
def remove_symbol(string, index):
    return string[:index] + string[index + 1:]


def check_edit(str1, str2):
    if abs(len(str1) -  len(str2)) >= 2:
        return False

    i = 0
    length = min(len(str1), len(str2))

    while i < length:
        if str1[i] == str2[i]:
            i += 1
        else:
            if len(str1) == len(str2):
                str1, str2 = remove_symbol(str1, i), remove_symbol(str2, i)
                length -= 1
            elif len(str1) > len(str2):
                str1 = remove_symbol(str1, i)
            elif len(str1) < len(str2):
                str2 = remove_symbol(str2, i)

            if str1 != str2:
                return False
    return True


def main():
    str1 = "spale"
    str2 = "spales"

    print(check_edit(str1, str2))


if __name__ == '__main__':
    main()
